internal/pkg/cli/env_init.go

```diff
@@ -206,9 +206,25 @@ func (o *initEnvOpts) Validate() error {
 			return err
 		}
 	}
+
 	if o.AppName() == "" {
-		return fmt.Errorf("no application found: run %s or %s into your workspace please", color.HighlightCode("app init"), color.HighlightCode("cd"))
+		existingApps, _ := o.store.ListApplications()
+		if len(existingApps) == 0 {
+			o.askAppName(fmtAppInitNamePrompt)
+		} else {
+			useExistingApp, err := o.prompt.Confirm(
+			"Would you like to use one of your existing applications?", "", prompt.WithTrueDefault(), prompt.WithFinalMessage("Use existing application:"))
+			if err != nil {
+				return fmt.Errorf("prompt to confirm using existing application: %w", err)
+			}
+			if useExistingApp {
+				o.askSelectExistingAppName(existingApps)
```

Comment:
Can we move this code to the `Ask()` method? Typically we reserve `Validate()` for read-only validation of the existing parameters. So for instance, in `Validate()` we could check whether an app exists in the workspace:

```go
if o.AppName() == "" {
    existingApps, _ := o.store.ListApplications()
    if len(existingApps) == 0 {
        return fmt.Errorf("no application found: run %s or %s into your workspace please", color.HighlightCode("app init"), color.HighlightCode("cd"))
    }
```

and then in `Ask()` we can run the code to prompt the customer for an existing app:
```go
if o.AppName == "" {
    existingApps, _ := o.store.ListApplications()
    o.askSelectExistingAppname(existingApps)
}
```